ext/spl/spl_iterators.c

```diff
@@ -1866,6 +1866,9 @@ PHP_METHOD(RegexIterator, accept)
 			}
 
 			result = php_pcre_replace_impl(intern->u.regex.pce, subject, ZSTR_VAL(subject), ZSTR_LEN(subject), replacement_str, -1, &count);
+			if (UNEXPECTED(!result)) {
+				RETURN_THROWS();
```

Comment:
`RETURN_THROWS()` expects an exception to have already been thrown, what is not (necessarily) the case here.